tests/query/test_col_replacement.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+
+from typing import Any, Mapping, MutableMapping
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.query import Query
+
+test_data = [
+    (
+        # Basic one column change
+        {
+            "selected_columns": ["c1", "c2", ["f1", ["c1"]]],
+            "conditions": [["c1", "=", "a"]],
+        },
+        "c1",
+        "c3",
+        {
+            "selected_columns": ["c3", "c2", ["f1", ["c3"]]],
+            "conditions": [["c3", "=", "a"]],
+        }
+    ),
+    (
+        # Test simple change in all fields
+        {
+            "selected_columns": ["c1", "c2", "c3"],
+            "conditions": [["c1", "=", "a"]],
+            "arrayjoin": "c1",
+            "groupby": "c1",
+            "aggregations": [["count()", ["c1"], "c1"]],
```

Comment:
This looks like a cross between the 2 syntaxes for aggregation. (The syntax that passes arguments as a list shouldn't have the parens?) Since `count()` doesn't take any argument it's also a weird example

Reply:
fixed